lib/js/src/manager/screen/utils/_PresentAlertOperation.js

```diff
@@ -0,0 +1,501 @@
+            alert.setAlertText3(this._alertView.getTertiaryText());
+        }
+
+        return alert;
+    }
+
+    /**
+     * Check to see if template supports soft button images.
+     * @returns {Boolean} - True if soft button images are supported, false if not.
+     */
+    supportsSoftButtonImages () {
+        const softButtonCapabilities = this._currentWindowCapability.getSoftButtonCapabilities()[0];
+        return softButtonCapabilities.getImageSupported();
+    }
+
+    /**
+     * Check to see if template supports Alert audio files.
+     * @returns {Boolean} - True if Alert audio files are supported, false if not.
+     */
+    supportsAlertAudioFile () {
+        return (this._lifecycleManager !== null && this._lifecycleManager !== undefined
+             && this._lifecycleManager.getSdlMsgVersion() !== null && this._lifecycleManager.getSdlMsgVersion() !== undefined
+             && this._lifecycleManager.getSdlMsgVersion().getMajorVersion() >= 5
+             && this._speechCapabilities !== null && this._speechCapabilities.includes(SpeechCapabilities.FILE));
+    }
+
+    /**
+     * Check to see if template supports Alert icons.
+     * @returns {Boolean} - True if Alert icons are supported, false if not.
+     */
+    supportsAlertIcon () {
+        return _ManagerUtility.hasImageFieldOfName(this._currentWindowCapability, ImageFieldName.alertIcon);
+    }
+
+    /**
+     * Check to see if template supports the soft button images.
+     * @param {AlertView} alertView - An AlertView instance.
+     * @returns {Boolean} - True if soft button images are supported, false if not.
+     */
+    isValidAlertViewData (alertView) {
+        if ((alertView.getText() !== null && alertView.getText() !== undefined && alertView.getText().length > 0)
+            || (alertView.getSecondaryText() !== null && alertView.getSecondaryText() !== undefined && alertView.getSecondaryText().length > 0)
+            || (alertView.getAudio() !== null && alertView.getAudio() !== undefined && alertView.getAudio().getAudioData().length > 0)) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Method to be called once the task has completed.
+     * @private
+     * @param {Boolean} success - Whether the task was successful.
+     * @param {Number} tryAgainTime - Try again time.
+     */
+    _finishOperation (success, tryAgainTime) {
+        console.log('Finishing present alert operation');
+        if (this._listener !== null && this._listener !== undefined) {
+            this._listener.onComplete(success, tryAgainTime);
+        }
+        this.onFinished();
+    }
+
+    /**
+     * Updates WindowCapability if the operation is pending the in the Alert Manager.
+     * @param {WindowCapability} defaultMainWindowCapability - The capabilities of the default main window.
+     */
+    setWindowCapability (defaultMainWindowCapability) {
+        this._currentWindowCapability = defaultMainWindowCapability;
+    }
```

Comment:
I found that I did not actually need this and could assign the WindowCapability in BaseAlertManager like 

```
((PresentAlertOperation) task).currentWindowCapability = currentWindowCapability;
```